app/views/contexts/summary/answer.py

```diff
@@ -86,16 +88,24 @@ def _return_to_answer_id(
         return_to: str | None,
         list_item_id: str | None,
         is_in_repeating_section: bool,
+        return_to_answer_id: str | None,
     ) -> str | None:
         """
         If the summary page using this answer has repeating answers, but it is not in a repeating section,
         then the answer ids will be suffixed with list item id, so the return to answer id link also needs this to work correctly
         """
+        answer_id = None
         if return_to:
             if (
                 list_item_id
                 and not is_in_repeating_section
                 and not self._original_answer_id  # original answer would mean id has already been suffixed
             ):
-                return f"{self.id}-{list_item_id}"
-            return self.id
+                answer_id = f"{self.id}-{list_item_id}"
+            else:
+                answer_id = self.id
+
+        if return_to_answer_id:
+            # Type ignore: return_to_answer_id will always be populated at this point
```

Comment:
The type ignore is for the wrong reasons, this will be complaining because `answer_id` could be None
You can fix this if you nest this block inside the `if return_to` 
it will only be populated if `return_to` is also populated so it makes sense to nest it and fixes the type hinting